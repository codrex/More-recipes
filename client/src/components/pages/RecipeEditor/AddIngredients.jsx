import React from 'react';
import { reduxForm } from 'redux-form';
import { connect } from 'react-redux';
import AddItems from './AddItems';
import {
  updateIngredients,
  clearValidationError
} from '../../../actions/recipeActions';

const AddIngredients = props => (
  <AddItems
    name="ingredient"
    ingredients
    placeholder="Enter ingredient"
    {...props}
  />
);

const mapStateToProps = state => (
  {
    items: state.recipe.ingredients,
    externalError: state.recipeValidationError.ingredients,
    message: state.networkRequest.msg
  }
);

export default reduxForm({
  form: 'ingredientForm'
})(connect(mapStateToProps, {
  sendItemsToStore: updateIngredients,
  clearValidationError
})(AddIngredients));
