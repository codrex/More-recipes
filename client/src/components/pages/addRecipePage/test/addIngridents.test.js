import React from 'react';
import AddIngredients from '../addIngredients/addIngredients';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from '../../../../store/configStore';


const props = {
  clearValidationError: jest.fn(),
  directions: true,
  externalError: [],
  handleSubmit: jest.fn(),
  ingredients: false,
  initialize: jest.fn(),
  items: [],
  name: '',
  placeholder: '',
  sendItemsToStore: jest.fn(),
}

describe('Add ingredients component ', () => {
  test('render as expected', () => {
    const component = renderer.create(
      <Provider store={configureStore()}>
        <AddIngredients {...props}/>
      </Provider>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
