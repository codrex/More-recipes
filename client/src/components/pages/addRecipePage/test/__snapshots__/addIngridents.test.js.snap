// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add ingredients component  render as expected 1`] = `
<div
  className="col-12 items "
>
  <div
    className="items-header"
  >
    <h4
      className="items-header-text"
    >
      s  0
    </h4>
    <form
      action=""
      className="col-12"
      method=""
      onSubmit={[Function]}
    >
      <div
        className="form-group d-flex flex-column no-margin"
      >
        <textarea
          aria-describedby="Textarea"
          className="form-control text-area no-margin add-item-input"
          id="Textarea"
          name=""
          onBlur={[Function]}
          onChange={[Function]}
          onDragStart={[Function]}
          onDrop={[Function]}
          onFocus={[Function]}
          placeholder=""
          rows="1"
          value=""
        />
        <div
          className=""
        />
      </div>
      <button
        className="btn text-capitalize w-100 btn-secondary"
        disabled={false}
        id="submit"
        type="submit"
      >
        add 
      </button>
    </form>
  </div>
  <div
    className="col-12 items-list-wrapper"
  >
    <div />
  </div>
</div>
`;
